samples/unit-tests/indicator-volume-by-price/recalculations/demo.js

```diff
@@ -154,6 +154,24 @@ QUnit.test('Test algorithm on data updates.', function (assert) {
 
     // End of #16277 testing, start other tests
 
+    VBPIndicator.update({
+        params: {
+            ranges: 6
+        }
+    });
+
+    assert.strictEqual(
+        VBPIndicator.options.params.ranges,
+        6,
+        'We should be able to update VBP params with update(), #17007.'
```

Comment:
The test looks absolutely fine, just two minor suggestions:

- Maybe we also should check for the `volumeSeriesData.length` since it was causing the problem?
`volumeSeriesData.length === VBPIndicator.options.params.ranges` should do the job.

- I would suggest changing the description to something more consistent with other tests, i.e:
"VBP params should be updated after update(), #17007"